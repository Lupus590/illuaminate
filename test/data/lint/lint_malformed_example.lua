--- config: (lint (only doc:malformed-example))
-- Checks malformed examples and other code blocks
-- @module Test


local x = {}

--- Descriptions which work
-- ```lua
-- return 0 -- statement
-- ```
-- ```lua
-- 0 -- expr
-- ```
--- ```lua
-- local more_complex
-- print(example)
-- ```
function x.description_pass() end

--- Descriptions which don't work
-- ```lua
-- Lua fenced code block
-- ```
--
--     Code block, but not marked as Lua
--
-- ```
-- Plain text code block
-- ```
function x.description_fail() end

--- @usage Single-line usage
-- @usage Multiline-code block usage
-- ```lua
-- Fenced in example
-- ```
--
--     Indented in example
function x.usage() end

return x
