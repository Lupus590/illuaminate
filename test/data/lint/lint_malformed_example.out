lint_malformed_example.lua:[31:1-31:33]: Cannot parse example [doc:malformed-example]

  =input:[1:5-1:10]: Unexpected `fenced`: Unexpected token after variable. [parse:syntax-error]
     │
   1 │ Lua fenced code block
     │     ^^^^^^
  
    │
 31 │ function x.description_fail() end
    │ ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
lint_malformed_example.lua:[40:1-40:22]: Cannot parse example [doc:malformed-example]

  =input:[1:10-1:11]: Unexpected `in`: Unexpected token after variable. [parse:syntax-error]
     │
   1 │ Indented in example
     │          ^^
  
    │
 40 │ function x.usage() end
    │ ^^^^^^^^^^^^^^^^^^^^^^
lint_malformed_example.lua:[40:1-40:22]: Cannot parse example [doc:malformed-example]

  =input:[1:8-1:9]: Unexpected `in`: Unexpected token after variable. [parse:syntax-error]
     │
   1 │ Fenced in example
     │        ^^
  
    │
 40 │ function x.usage() end
    │ ^^^^^^^^^^^^^^^^^^^^^^
lint_malformed_example.lua:[40:1-40:22]: Cannot parse example [doc:malformed-example]

  =input:[1:7-1:7]: Unexpected `-`: Unexpected token after variable. [parse:syntax-error]
     │
   1 │ Single-line usage
     │       ^
  
    │
 40 │ function x.usage() end
    │ ^^^^^^^^^^^^^^^^^^^^^^
No errors and 4 warnings
